src/notevennexus/validators.py

```diff
@@ -0,0 +1,269 @@
+
+    def validate(self, node: Dataset | Group) -> Violation | None:
+        if "NX_class" not in node.attrs:
+            return Violation(node.name)
+
+
+class depends_on_target_missing(Validator):
+    def __init__(self) -> None:
+        super().__init__("depends_on_target_missing", "depends_on target is missing")
+
+    def applies_to(self, node: Dataset | Group) -> bool:
+        return node.name.endswith('/depends_on') or "depends_on" in node.attrs
+
+    def validate(self, node: Dataset | Group) -> Violation | None:
+        if node.name.endswith('/depends_on'):
+            target = node.value
+        else:
+            target = node.attrs["depends_on"]
+        if target == '.':
+            return None
+        if not isinstance(target, str):
+            return Violation(node.name, f"depends_on target {target} is not a string")
+        path = target.split('/')
+        if path[0] == '':
+            start = self._find_root(node)
+            path = path[1:]
+        else:
+            start = node.parent
+        if path[0] == '.':
+            path = path[1:]
+        for name in path:
+            if name not in start.children:
+                return Violation(node.name, f"depends_on target {target} is missing")
+            start = start.children[name]
+
+    def _find_root(self, node: Dataset | Group) -> Dataset | Group:
+        while node.parent is not None:
+            node = node.parent
+        return node
+
+
+class NX_class_is_legacy(Validator):
+    def __init__(self) -> None:
+        super().__init__("NX_class_is_legacy", "Check if NX_class is deprecated")
+
+    def applies_to(self, node: Dataset | Group) -> bool:
+        return isinstance(node, Group) and "NX_class" in node.attrs
+
+    def validate(self, node: Dataset | Group) -> Violation | None:
+        nx_class = node.attrs.get("NX_class")
+        if nx_class in ['NXgeometry', 'NXorientation', 'NXshape', 'NXtranslation']:
+            return Violation(node.name, f"NX_class {nx_class} is deprecated")
+
+
+class group_has_units(Validator):
+    def __init__(self) -> None:
+        super().__init__("group_has_units", "Group should not have units attribute")
+
+    def applies_to(self, node: Dataset | Group) -> bool:
+        return isinstance(node, Group)
+
+    def validate(self, node: Dataset | Group) -> Violation | None:
+        if 'units' in node.attrs:
+            return Violation(node.name)
+
+
+class units_invalid(Validator):
+    def __init__(self) -> None:
+        super().__init__("units_invalid", "Invalid units attribute")
+
+    def applies_to(self, node: Dataset | Group) -> bool:
+        return isinstance(node, Dataset) and 'units' in node.attrs
+
+    def validate(self, node: Dataset | Group) -> Violation | None:
+        units = node.attrs['units']
+        if units.startswith('NX_'):  # Placeholder from NeXus standard
+            return Violation(node.name, f"Invalid units {units}")
+        invalid = ['hz']
```

Comment:
Would be nice to be able to go further, maybe trying to parse it with Scipp, but then scipp would become a dependency, and that's not desirable for such a tool...

Reply:
I was considering to have an option:
```sh
notevennexus --use-scipp file.nxs
```
or something like that to enable some checks like you suggest.

Reply:
The other idea was to look for a table of "valid" units somewhere...